### ISSUE TITLE
CheckBox

### ISSUE THREAD

**Issue by panliang-rm** [AmrDeveloper/TreeView#25]
Whether i can check the parent node, all its child nodes will be checked.
![treeView](https://github.com/AmrDeveloper/TreeView/assets/54338815/f9130e3f-6545-4945-acec-5d77e1434f2f)

**Comment by AmrDeveloper** [comment]
Hello @panliang-rm,

Sorry for delay, do you still have a code snippets on how to reproduce it

**Comment by panliang-rm** [comment]
Hello, this is an example I found online. I want to ask if TreeView can provide checkbox functionality. Currently, my implementation is quite complex and not elegant - I need to manually traverse the data source and use EventBus to notify updates.

**Comment by AmrDeveloper** [comment]
@panliang-rm Yes, TreeView allow you to define your custom view with custom logic too, I created a small example with CheckBox component

https://github.com/AmrDeveloper/TreeView/tree/master/app/src/main/java/com/amrdeveloper/treeviewlib/todotree


https://github.com/user-attachments/assets/5f98123c-d0a2-4ba4-90d5-65c0cbb094fe

You can also decide if you want to check the parent if all children are checked or not
